flutter-idea/src/io/flutter/analytics/UnifiedAnalytics.java

```diff
@@ -0,0 +1,232 @@
+      assert result != null;
+      JsonPrimitive value = result.getAsJsonPrimitive("value");
+      CompletableFuture<Boolean> innerResult = new CompletableFuture<>();
+
+      if (value == null) {
+        return CompletableFuture.failedFuture(new Exception("telemetryEnabled value is null"));
+      }
+      innerResult.complete(value.getAsBoolean());
+      return innerResult;
+    });
+  }
+
+  private @Nullable CompletableFuture<Boolean> clientShowedMessage(@NotNull DartToolingDaemonService service, @NotNull JsonObject params) {
+    return makeUnifiedAnalyticsRequest("clientShowedMessage", service, params).thenCompose(result -> {
+      assert result != null;
+      JsonPrimitive type = result.getAsJsonPrimitive("type");
+      CompletableFuture<Boolean> innerResult = new CompletableFuture<>();
+
+      if (type == null) {
+        return CompletableFuture.failedFuture(new Exception("clientShowedMessage type is null"));
+      }
+      innerResult.complete("Success".equals(type.getAsString()));
+      return innerResult;
+    });
+  }
+
+  private @Nullable CompletableFuture<Boolean> setTelemetry(@NotNull DartToolingDaemonService service, @NotNull JsonObject params, Boolean canSendAnalytics) {
+    params.addProperty("enable", canSendAnalytics);
+    return makeUnifiedAnalyticsRequest("setTelemetry", service, params).thenCompose(result -> {
+      assert result != null;
+      JsonPrimitive type = result.getAsJsonPrimitive("type");
+      CompletableFuture<Boolean> innerResult = new CompletableFuture<>();
+
+      if (type == null) {
+        return CompletableFuture.failedFuture(new Exception("setTelemetry type is null"));
+      }
+      innerResult.complete("Success".equals(type.getAsString()));
+      return innerResult;
+    });
+  }
+
+  private CompletableFuture<Boolean> showMessage(@NotNull String message) {
+    CompletableFuture<Boolean> finalResult = new CompletableFuture<>();
+    ApplicationManager.getApplication().invokeLater(() -> {
+      final Notification notification = new Notification(
+        Analytics.GROUP_DISPLAY_ID,
+        "Welcome to Flutter!",
+        message,
+        NotificationType.INFORMATION);
+      //noinspection DialogTitleCapitalization
+      notification.addAction(new AnAction("Sounds good!") {
+        @Override
+        public void actionPerformed(@NotNull AnActionEvent event) {
+          notification.expire();
+          finalResult.complete(true);
+        }
+      });
+      //noinspection DialogTitleCapitalization
+      notification.addAction(new AnAction("No thanks") {
+        @Override
+        public void actionPerformed(@NotNull AnActionEvent event) {
+          notification.expire();
+          finalResult.complete(false);
+        }
+      });
+      Notifications.Bus.notify(notification, project);
+      System.out.println("Ran notification");
+    });
+    return finalResult;
+  }
+
+  private String getToolName() throws Exception {
```

Comment:
could be beneficial to add a link to the unified_analytics package library where these values are defined so that it is clear where the strings came from.

Reply:
Definitely, I'll add this in a next PR